WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/DashboardPostsListCardCell.swift

```diff
@@ -103,23 +99,35 @@ extension DashboardPostsListCardCell {
             assertionFailure("Cell used with wrong card type")
             return
         }
+        addContextMenu(card: cardType, blog: blog)
+
         viewModel = PostsCardViewModel(blog: blog, status: status, view: self)
         viewModel?.viewDidLoad()
         tableView.dataSource = viewModel?.diffableDataSource
         viewModel?.refresh()
     }
 
+    private func addContextMenu(card: DashboardCard, blog: Blog) {
+        frameView.onEllipsisButtonTap = {
```

Comment:
I'm curious about your thoughts.

> Now every card from the new "Personalize Home Tab" screen has a "Hide this" button in the context menu. 

I'm looking at each of the cells which repeat the same pattern of adding ellipsis button handling, hiding action, and tracking code. 

Although duplication is not necessarily evil, I'm wondering if in this situation it makes and is possible to have something reused. To describe the card as "personalizable" and then it will share common these functionality and layout traits.

What do you think about that?

>A couple more cards require special attention and will be made "personalizable" later.

Is this a counter-argument to my question and it's difficult to reuse the same layout and functionality code between personalizable cards?

Reply:
Thanks for the insightful answer and your examples! Those `extensions` look like good wins to take. 

> Should I make this change in this PR? I have some other ideas for refactoring this and also Dashboard.personalizableCards. I was going to open a new PR after getting the features merged first.

I think we could do that in other PRs as well. There are already a fair amount of changes, it will be easier to digest additional ones in a separate PR.